src/fswatch_win/fswatch_win_stubs.c

```diff
@@ -0,0 +1,430 @@
+/* File-watching support under Windows
+
+   We implement file-watching support under Windows using ReadDirectoryChangesW
+   and I/O completion ports. The notification of events is pull-based (one needs
+   to retrive them explicitly, there is no callback that is invoked on each
+   notification).
+
+   The main type is [fsenv] which keeps track of the state of a file watcher
+   (including all changes that have been notified by the operator system, but
+   that have not yet been retrieved by the user).
+
+   There is one value of type [wenv] for each directory being watched. It keeps
+   track of the various buffers needed to use the ReadDirectoryChangesW function
+   call.
+
+   When the first watch is created a native (not OCaml) thread is triggered
+   which runs the main event loop, waiting for notifications from the I/O
+   completion port. When a notification arrives, it is recorded in the [fsenv]
+   and it will stay there until the user retrives it by calling [wait].  We
+   notify the user that new notifications have arrived by signalling an event
+   object which causes the [wait] function to return.
+
+   Available notifications can be retrieved by calling [get_events]. The
+   [.events] field of the [fsenv] type which is mutated from multiple threads.
+   Standard CAS techniques are used to avoid data races.
+ */
+
+#define CAML_NAME_SPACE
+
+#include <caml/mlvalues.h>
+#include <caml/alloc.h>
+#include <caml/fail.h>
+#include <caml/osdeps.h>
+#include <caml/memory.h>
+#include <caml/threads.h>
+#include <caml/unixsupport.h>
+
+#if OCAML_VERSION_MAJOR < 5
+#define caml_uerror uerror
+#define caml_win32_maperr win32_maperr
+#endif
+
+#if 0
+#define DEBUG(fmt, ...) \
+  do { fprintf(stderr, "DEBUG: " __FUNCTION__ ": %d: " fmt "\n", __LINE__, __VA_ARGS__); fflush(stderr); } while (0)
+#else
+#define DEBUG(fmt, ...)
```

Comment:
```suggestion
#define DEBUG(fmt, ...) ((void) 0)
```

This avoids problem that may result from the macro expanding to nothing, and the semicolon after the macro use acting on code before. It's not the case in the current code, but that may happen in the future.